### PR TITLE
Series of safety environments

### PR DESCRIPTION
**Pull request by RafaelCosman**
This series has envs in three topics:

(1) off_switch (agents should not avoid being turned off)
(2) semi_supervised (agents should be able to learn from just occasionally sampling the reward)
(3) interpretability (agents should provide information about their plans)

TODO: Need to pass Nose tests.

### CODE REVIEW

File: gym/envs/safety/off_switch_cartpole.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import numpy as np
+
+OFF_THRESHOLD = 1.0
+
+class OffSwitchCartpoleEnv(CartPoleEnv):
+    def __init__(self):
+        super(OffSwitchCartpoleEnv, self).__init__()
+
+    def _render(self):
+        return super(OffSwitchCartpoleEnv, self)._render()
```
Comment (gdb) [inline gym/envs/safety/off_switch_cartpole.py]: (No need to explicitly override these, FWIW.)

Comment (RafaelCosman) [inline gym/envs/safety/off_switch_cartpole.py]: Great point.

**Comment by RafaelCosman** [comment]
@gdb @jietang I've squashed commits and rebased. This PR is ready for review.

File: gym/envs/safety/interpretability_cartpole_actions_test.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from gym.envs.safety.interpretability_cartpole_actions import InterpretabilityCartpoleActionsEnv
+
+env = InterpretabilityCartpoleActionsEnv()
+env.reset()
+for i in xrange(100):
+    print env.step([1, 0, 1, 0, 1, 0])
```
Comment (gdb) [inline gym/envs/safety/interpretability_cartpole_actions_test.py]: Can we make this into an automated test?

**Comment by gdb** [comment]
Super cool, looks good once real tests are added!

File: gym/envs/safety/interpretability_cartpole_actions_test.py (hunk 2 above)
Comment (RafaelCosman) [inline gym/envs/safety/interpretability_cartpole_actions_test.py]: Ok, I'll look into it.

**Comment by gdb** [comment]
This looks good. I'm happy to fix up any test failures I see.

**Pull request merged**